src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,46 @@
 package practice;
 
-public class CandidateValidator {
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE = 35;
+    private static final String NATIONALITY = "Ukrainian";
+    private static final int YEARS_IN_COUNTRY = 10;
+    private static final int PERIOD = 2;
+    private static final String PATTERN_TIME = "[0-9]{4}";
+    private Matcher matcher;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate != null && candidate.getAge() >= MIN_AGE
+                && NATIONALITY.equals(candidate.getNationality())
+                && candidate.isAllowedToVote() && checkPeriod(candidate.getPeriodsInUkr());
+    }
+
+    private boolean checkPeriod(String period) {
+        if (period == null || isMatch(period)) {
+            return false;
+        }
+        int[] periods = new int[PERIOD];
+        matcher = Pattern.compile(PATTERN_TIME).matcher(period);
+        for (int i = 0; i < PERIOD; i++) {
+            if (matcher.find()) {
+                periods[i] = Integer.parseInt(matcher.group());
+            }
+        }
+        return Math.abs(periods[0] - periods[1]) >= YEARS_IN_COUNTRY;
```

Comment:
ми можемо спростити цей метод до:
```suggestion
        if (period == null || ! period.matches(PATTERN_TIME)) {
            return false;
        }
        String[] data = period.split("-");
        return Integer.valueOf(data[0]) - Integer.valueOf(data[1]) >= YEARS_IN_COUNTRY;
```

Reply:
```suggestion
! period.matches(PATTERN_TIME))
```
Это так не работает.
И вся структура валиться в случаях:
Если нету разделителя "-" (в моем случае мне он не нужен, я мог вообще выловить что-то аля 20102020).
И если дата не полностью (205-2020) -  я такое даже не допускал, а в этом случае все сработает.
Оставил с твоими пожеланиями.
C наступающим тебя ;)

Reply:
Саша, завдання призначене для того щоб ви навчились використовувати Predicate а не для того щоб навчитись валідувати дані. Загалом можна розглядати величезну кількість варіантів вхідних даних, і писати складні валідатори для кожного завдання. 
Так як ти вирішив, все ж таки, валідувати дані тут - я хотіла показати спосіб як це можна зробити. Можеш залишити свій варіант, якщо він тобі більше подобається) It's up to you.

Reply:
Дякую, і тебе з наступаючим 🌲